I tried this on `x86_64-linux` and both the Wayland and X variants appear to work, Sway however complains about the lack of hardware rendering:

```
machineNoX11 # [   39.108042] xsession[978]: 00:00:00.366 [ERROR] [wlr] [render/egl.c:334] Software rendering detected, please use the WLR_RENDERER_ALLOW_SOFTWARE environment variable to proceed
machineNoX11 # [   39.110573] xsession[978]: 00:00:00.369 [ERROR] [wlr] [render/egl.c:611] Failed to initialize EGL context
machineNoX11 # [   39.112389] xsession[978]: 00:00:00.369 [ERROR] [wlr] [render/gles2/renderer.c:499] Could not initialize EGL
machineNoX11 # [   39.346539] xsession[978]: 00:00:00.605 [ERROR] [wlr] [render/vulkan/renderer.c:2516] Could not match drm and vulkan device
```

A screenshot is created successfully nonetheless.

--- a/nixos/tests/ly.nix
+++ b/nixos/tests/ly.nix
@@ -36,27 +36,56 @@ in
     let
       user = nodes.machine.users.users.alice;
     in
+    # python
     ''
+      from test_driver.errors import RequestedAssertionFailed
       start_all()
 
-      machine.wait_until_tty_matches("1", "password:")
+      # old ly versions before 1.1.2 used to allow typing the username
+      # but now a user can only be selected from a set of users
+      def navigate_user(machine, username, wm, tty="1"):
+          wm = wm.lower()
+          tries = 0
+          while username not in machine.get_tty_text(tty):
+              machine.send_key("left")
+              machine.sleep(0.3)
+              if tries > 3:
+                  RequestedAssertionFailed(f"Failed to find user:{username} in ly")
+              tries += 1
+
+          # move cursor to wm selection
+          machine.send_key("up")
+
+          tries = 0
+          while wm not in machine.get_tty_text(tty).lower():
+              machine.send_key("left")
+              machine.sleep(0.3)
+              if tries > 3:
+                  RequestedAssertionFailed(f"Failed to find wm:{wm} in ly")
+              tries += 1
+
+          # reset cursor to user selection
+          machine.send_key("tab")
+
+      machine.wait_until_tty_matches("1", "password")
       machine.send_key("ctrl-alt-f1")
       machine.sleep(1)
       machine.screenshot("ly")
-      machine.send_chars("alice")
+      navigate_user(machine, "${user.name}", "icewm")
       machine.send_key("tab")
       machine.send_chars("${user.password}")
       machine.send_key("ret")
       machine.wait_for_file("/run/user/${toString user.uid}/lyxauth")
       machine.succeed("xauth merge /run/user/${toString user.uid}/lyxauth")
       machine.wait_for_window("^IceWM ")
+      machine.sleep(2)
       machine.screenshot("icewm")
 
-      machineNoX11.wait_until_tty_matches("1", "password:")
+      machineNoX11.wait_until_tty_matches("1", "password")
       machineNoX11.send_key("ctrl-alt-f1")
       machineNoX11.sleep(1)
       machineNoX11.screenshot("ly-no-x11")
-      machineNoX11.send_chars("alice")
+      navigate_user(machineNoX11, "${user.name}", "Sway")
       machineNoX11.send_key("tab")
       machineNoX11.send_chars("${user.password}")
       machineNoX11.send_key("ret")

--- a/nixos/tests/ly.nix
+++ b/nixos/tests/ly.nix
@@ -67,6 +67,8 @@ in
           # reset cursor to user selection
           machine.send_key("tab")
 
+      # https://github.com/NixOS/nixpkgs/pull/455191#discussion_r2507716719
+      machine.wait_until_succeeds("getfacl /dev/dri/card0 | grep video")
       machine.wait_until_tty_matches("1", "password")
       machine.send_key("ctrl-alt-f1")
       machine.sleep(1)

